File: tests/rptest/services/chaos/workloads/list_offsets/service.py
# Copyright 2024 Redpanda Data, Inc.
#
# Use of this software is governed by the Business Source License
# included in the file licenses/BSL.md
#
# As of the Change Date specified in that file, in accordance with
# the Business Source License, use of this software will be governed
# by the Apache License, Version 2.0

from ..service_base import WorkloadServiceBase
from . import consistency, stat


class ListOffsetsWorkload(WorkloadServiceBase):
    def __init__(self, ctx, brokers_str, topic, concurrency=2):
        super().__init__(ctx, brokers_str, num_nodes=1)
        self._topic = topic
        self._concurrency = concurrency

    @property
    def java_module_name(self):
        return "list_offsets"

    def extra_config(self, node):
        return {
            "topic": self._topic,
            "settings": {
                "concurrency": self._concurrency,
            }
        }

    def validate_consistency(self):
        consistency.validate(
            self._results_dir(),
            [n.name for n in self.nodes],
            brokers_str=self._brokers_str,
            topic=self._topic,
        )

    def collect_stats(self):
        return stat.collect(self.context.test_id, self._results_dir(),
                            [n.name for n in self.nodes])

File: src/v/crash_tracker/service.cc
/*
 * Copyright 2025 Redpanda Data, Inc.
 *
 * Use of this software is governed by the Business Source License
 * included in the file licenses/BSL.md
 *
 * As of the Change Date specified in that file, in accordance with
 * the Business Source License, use of this software will be governed
 * by the Apache License, Version 2.0
 */

#include "crash_tracker/service.h"

#include "config/node_config.h"
#include "crash_tracker/limiter.h"
#include "crash_tracker/recorder.h"

#include <seastar/core/seastar.hh>

namespace crash_tracker {

service::service() noexcept
  : _limiter(get_recorder()) {}

ss::future<> service::start() {
    if (config::node().developer_mode()) {
        // crash loop tracking has value only in long running clusters
        // that can potentially accumulate state across restarts.
        co_return;
    }

    co_await _limiter.check_for_crash_loop();
    co_await get_recorder().start();
}

ss::future<> service::stop() {
    co_await _limiter.record_clean_shutdown();
    co_await get_recorder().stop();
}

} // namespace crash_tracker
